src/lsdb/dask/crossmatch_catalog_data.py

```diff
@@ -84,28 +102,97 @@ def perform_crossmatch(
     npd.NestedFrame
         DataFrame with the results of crossmatching for the pair of partitions.
     """
-    if right_pix.order > left_pix.order:
+    # If there's no left partition for this aligned pixel, return an empty/meta frame
+    if left_df is None or left_df.size == 0:
```

Comment:
A nit-picky suggestion since `DataFrame.size()` calculates `num_rows*num_cols`:
```suggestion
    if left_df is None or len(left_df) == 0:
```